### PR TITLE
Reinstate progress bars and cleanup unused code/dependencies

### PR DESCRIPTION
**Pull request by rtibbles**
### Summary
* Takes advantage of the introduction of `Click` to remove our use of `clint`
* Adds progress bars back in to our base async management command, but only if called inside a `Click` cli context
* Deletes the previously deprecated syncdata command
* Deletes the unused `search.py` module
* Removes unused dependencies resulting from deleted or edited code

### Reviewer guidance
Does everything still work?
Do the progress bars only appear on the command line?

----

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/core/deviceadmin/management/commands/dbrestore.py [hunk 1]
```diff
@@ -92,19 +92,19 @@ def select_backup(self, dumps_root):
         if not backups:
             raise RuntimeError("Could not find a database backup}")
```
Comment (indirectlylit) [inline kolibri/core/deviceadmin/management/commands/dbrestore.py]: This wasn't part of your PR, but hitting this `RuntimeError` is a poor experience for a non-exceptional scenario. Perhaps just log the message and exit?

File: kolibri/core/deviceadmin/management/commands/dbrestore.py [hunk 2]
```diff
@@ -92,19 +92,19 @@ def select_backup(self, dumps_root):
...
+            "Type the number in brackets to select the backup to be restored\n"
+            + "".join(
+                (
+                    "({num}) {backup}\n".format(
+                        num=num + 1, backup=get_dtm_from_backup_name(backup)
+                    )
+                    for num, backup in enumerate(backups)
+                )
+            ),
+            type=click.Choice(range(1, len(backups) + 1)),
```
Comment (indirectlylit) [inline kolibri/core/deviceadmin/management/commands/dbrestore.py]: Running Python 3.7.3 I get

```
  ...

  File "/Users/d/Projects/le/kolibri/kolibri/core/deviceadmin/management/commands/dbrestore.py", line 105, in select_backup
    type=click.Choice(range(1, len(backups) + 1)),
  File "/Users/d/Projects/le/kolibri/.venv/lib/python3.7/site-packages/click/termui.py", line 126, in prompt
    result = value_proc(value)
  File "/Users/d/Projects/le/kolibri/.venv/lib/python3.7/site-packages/click/types.py", line 39, in __call__
    return self.convert(value, param, ctx)
  File "/Users/d/Projects/le/kolibri/.venv/lib/python3.7/site-packages/click/types.py", line 180, in convert
    (value, ', '.join(self.choices)), param, ctx)
TypeError: sequence item 0: expected str instance, int found
```

File: kolibri/core/tasks/management/commands/base.py [hunk 3]
```diff
@@ -22,7 +23,17 @@ def __init__(self, total=100, level=0, update_callback=None):
...
+            # Check we are executing inside a click context
+            # as we only want to display progress bars from the command line
+            click.get_current_context()
+            self.progressbar = click.progressbar(length=total, width=0)
+        except RuntimeError:
+            self.progressbar = None
+
     def update_progress(self, increment=1, message="", extra_data=None):
+        if self.progressbar:
+            self.progressbar.update(increment)
```
Comment (indirectlylit) [inline kolibri/core/tasks/management/commands/base.py:36]: neat!
image


doesn't do a great job of handling terminal resizes in mac, but that's not a deal-breaker:

image

**Review by indirectlylit — APPROVED** [review]
looks good!

couple minor things in the `dbrestore` command but otherwise good to go

File: kolibri/core/tasks/management/commands/base.py (hunk 3 above)
Comment (rtibbles) [reply kolibri/core/tasks/management/commands/base.py:36]: Yeah - I couldn't work out whether that was my fault or Click's.

File: kolibri/core/deviceadmin/management/commands/dbrestore.py (hunk 1 above)
Comment (rtibbles) [reply kolibri/core/deviceadmin/management/commands/dbrestore.py]: Seems reasonable.

File: kolibri/core/deviceadmin/management/commands/dbrestore.py (hunk 2 above)
Comment (rtibbles) [reply kolibri/core/deviceadmin/management/commands/dbrestore.py]: Will test locally!

**Pull request merged**